File: javascript/1700.Number-of-Students-Unable-to-Eat-Lunch.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function (students, sandwiches) {
  const map = new Map();
  let res = students.length;

  for (const s of students) {
    map.set(s, (map.get(s) || 0) + 1);
  }

  for (const s of sandwiches) {
    if (map.get(s) > 0) {
      res -= 1;
      map.set(s, map.get(s) - 1);
    } else {
      break;
    }
  }

  return res;
};

File: python/1700.Number-of-Students-Unable-to-Eat-Lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        freq = Counter(students)
        res = len(students)

        for s in sandwiches:
            if freq[s] > 0:
                res -= 1
                freq[s] -= 1
            else:
                break

        return res